src/fastmean.c

```diff
@@ -48,21 +48,24 @@ SEXP fastmean(SEXP args)
   if (narm) {
     switch(TYPEOF(x)) {
     case LGLSXP:
-    case INTSXP:
+    case INTSXP: {
+      int *pi = INTEGER(x);
       for (i = 0; i<l; i++) {
-        if(INTEGER(x)[i] == NA_INTEGER) continue;
-        s += INTEGER(x)[i];   // no under/overflow here, s is long double not integer
+        if(pi[i] == NA_INTEGER) continue;
+        s += pi[i];   // no under/overflow here, s is long double not integer
         n++;
       }
       if (n>0)
         REAL(ans)[0] = (double) (s/n);
       else
         REAL(ans)[0] = R_NaN;  // consistent with base: mean(NA,na.rm=TRUE)==NaN==mean(numeric(),na.rm=TRUE)
       break;
-    case REALSXP:
+    }
+    case REALSXP: {
+      double *pr = REAL(x);
       for (i = 0; i<l; i++) {
-        if(ISNAN(REAL(x)[i])) continue;  // TO DO: could drop this line and let NA propogate?
-        s += REAL(x)[i];
+        if(ISNAN(pr[i])) continue;  // TO DO: could drop this line and let NA propogate?
```

Comment:
`if` spacing, `continue`

comment could be removed, NAs should not propagate because it is `narm` branch